integration_test.go

```diff
@@ -321,21 +320,10 @@ func expectAndResetMetric(t *testing.T, metric *prometheus.CounterVec, expected
 	metric.Reset()
 }
 
-func makeTCH(url string, s3Service *s3.Client) tileCachingHandler {
-	return tileCachingHandler{
-		logURL:   url,
-		tileSize: 3,
-
-		s3Service: s3Service,
-		s3Prefix:  "test",
-		s3Bucket:  "bucket",
-
-		cacheGroup: &singleflight.Group{},
-
-		fullRequestTimeout: 10 * time.Second,
-
-		requestsMetric:     prometheus.NewCounterVec(prometheus.CounterOpts{Help: "foo", Name: "ctile_requests"}, []string{"result", "source"}),
-		partialTiles:       prometheus.NewCounter(prometheus.CounterOpts{Name: "ctile_partial_tiles"}),
-		singleFlightShared: prometheus.NewCounter(prometheus.CounterOpts{Name: "ctile_singleflight_shared"}),
+func makeTCH(url string, s3Service *s3.Client) *tileCachingHandler {
```

Comment:
You could add `t *testing.T` to the function signature and replace the only use of panic below with `t.Fatal(err)`. If you choose to do this, you'd need to also alter the two makeTCH calls up in `TestIntegration`.